pkg/netutil/netutil.go

```diff
@@ -0,0 +1,50 @@
+package netutil
+
+import (
+	"context"
+	"io"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+)
+
+func PublicIP() (string, error) {
+	// Create a transport that forces IPv4
+	transport := &http.Transport{
+		// Force IPv4 by using tcp4 network
+		DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+			return (&net.Dialer{
+				Timeout:   5 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext(ctx, "tcp4", addr)
+		},
+	}
+
+	client := &http.Client{
+		Timeout:   10 * time.Second,
+		Transport: transport,
+	}
+
+	// Create a request to mimic curl behavior
+	req, err := http.NewRequest("GET", "https://ifconfig.me", nil)
```

Comment:
make `ifconfig.me` as a param, if the param is empty, use `ifconfig.me` by default.